### PR TITLE
 Answer to exam question with Streamlit 

### PR DESCRIPTION
**Pull request by Goldria**
В этом проекте реализовано приложение Streamlit для работы с нейросетью. Необходимо срочно узнать возможные ответы на вопросы по английскому тексту из ЕГЭ? Введите весь текст и вопрос в соответствующие окошки, чтобы получить ответ, который сгенерирует нейросеть.

Выполнила Гольденберг Дарья.

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    context_input = get_context()
+    question_input = get_question()
+
+    model_name = "deepset/roberta-base-squad2"
+
+    nlp = pipeline("question-answering",
+                   model=model_name,
+                   tokenizer=model_name
+                   )
```
Comment (SpirinEgor) [inline src/main.py:32]: Your task is to generate text from some inputs. In NLP you have different strategies to do it based on the trained model. You can read this article to understand the differences (https://huggingface.co/blog/how-to-generate).

To handle these strategies, there is a special method [`generate`](https://huggingface.co/docs/transformers/v4.25.1/en/main_classes/text_generation#transformers.GenerationMixin.generate) in 🤗.

There are a lot of parameters, but you only need a couple of them:
- `max_length` and `min_length` to control the size of generated text
- `num_beams` to control the size of beam search, a bigger value gives more accurate, but strict text
- `do_sample` activate sampling strategies that add a little bit of randomness, these make the text more alive.
- `top_p` and `top_k` control the randomness.

I suggest you provide a special tab in your app, where users may control these generation parameters and see different examples of question answers. Text me in telegram, if you have questions about parameters and which is better to use.

**Review by SpirinEgor — CHANGES_REQUESTED** [review]
Good and accurate solution 👍

Let's add the functionality to play with various text generation parameters. This will add an opportunity to generate strict text that directly answers the question or to generate a long detailed answer.

**Comment by Goldria** [comment]
Let me remind you that we discussed the problem with ```generate()``` in telegram :)

**Review by SpirinEgor — APPROVED** [review]
Nice and accurate solution 👍

**Pull request merged**